packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -196,7 +203,8 @@ export class CdkToolkit {
           usePreviousParameters: options.usePreviousParameters,
           progress: options.progress,
           ci: options.ci,
-          rollback: options.rollback,
+          rollback: shouldDisableRollback ? false : true,
```

Comment:
Since you moved this here, we can simplify further:

```suggestion
          rollback: options.rollback === undefined && options.hotswap ? false : options.rollback,
```

And get rid of the `shouldDisableRollback` variable.